src/pages/submit/grant.tsx

```diff
@@ -79,9 +80,30 @@ function parseStringsAsNumbers(grantRequest: GrantRequest) {
   }
 }
 
+function handleCancel() {
+  if ((window as any).routeUpdate) {
+    window.history.back()
+  } else {
+    navigate(locations.submit())
+  }
+}
+
 export default function SubmitGrant() {
   const t = useFormatMessage()
   const [account, accountState] = useAuthContext()
+  useEffect(() => {
+    const params = new URLSearchParams(location.search)
+    let category: NewGrantCategory | null = null
+    try {
+      category = toNewGrantCategory(params.get('category'))
+    } catch (error) {
+      console.error(error)
+    } finally {
+      patchGrantRequest({ category })
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I'd remove this and add the dependencies. Is params triggering a re-render?

Reply:
<img width="867" alt="image" src="https://user-images.githubusercontent.com/45410089/222793520-98327add-7ff9-46dc-8d34-6fdfc5738e41.png">

It suggests adding the patch function to the dependencies, which is not necessary. That's why I omitted it

Reply:
Don't you need to add `location.search` to the deps?